src/ctap/storage.rs

```diff
@@ -151,6 +155,8 @@ impl PersistentStore {
     ///
     /// This should be at most one instance of persistent store per program lifetime.
     pub fn new(rng: &mut impl Rng256) -> PersistentStore {
+        // This should ideally be a compile-time assert, but Rust doesn't have native support.
+        assert!(STORE_SIZE <= STORE_SIZE_LIMIT);
```

Comment:
Can we do this in a `#[test]` to avoid the runtime overhead? And if this predicate would still be false at runtime, would it lead to a memory safety issue, or just the kernel stopping the app due to the MPU catching a memory violation?

Reply:
It would be an issue to run without this check because the driver doesn't check that it's in bound (and the MPU won't help when talking to the NVMC). But actually this should be considered a bug of the driver. I think it's better to do this check in the driver rather than here, so I'm moved the test.

Regarding the runtime overhead, the current one is totally negligible because it's once per process lifetime. The one when the driver does the check is also negligible but occurs at each syscall. Note that we already to this check. Now instead of checking against each writeable region of the app we check against the storage.